supabase/migrations/20250228194403_add-public-profile-table.sql

```diff
@@ -0,0 +1,277 @@
+drop function if exists "wallet"."upsert_user_with_accounts"(user_id uuid, email text, email_verified boolean, accounts jsonb[]);
+
+create table "wallet"."public_profiles" (
+    "id" uuid not null default gen_random_uuid(),
+    "created_at" timestamp with time zone not null default now(),
+    "username" text not null,
+    "user_id" uuid
+);
+
+alter table "wallet"."public_profiles" enable row level security;
+
+alter table "wallet"."users" add column "profile_id" uuid;
```

Comment:
isn't this wrong direction of the relationship? don't we want leave the room for having multiple profiles later and thus it should be one to many relationship where one user can have more profiles